### ISSUE TITLE
normalization.energy_from_power does not use power.index.freq where available

### ISSUE THREAD

**Issue by kanderso-nrel** [NREL/rdtools#206]
**Is your feature request related to a problem? Please describe.**
`normalization.energy_from_power` calculates the difference between timestamps to determine the measurement period of the data.  I don't really have an issue with this when there is a more straightforward alternative of using the `power.index.freq` attribute (maybe it's a slight inefficiency) but in the case where the time series is a single value, the `np.diff` approach fails.  It would be nice to have a shortcut of using the stated period when it is available. 

**Full error message and traceback**
```python
In [1]: import rdtools
   ...: rdtools.__version__
   ...: 
Out[1]: '2.0.0-beta.0+147.g1cb2a08'

In [2]: import pandas as pd
   ...: power = pd.Series([100], index=pd.date_range('2019-01-01', freq='15T', periods=1))
   ...: power
   ...: 
Out[2]: 
2019-01-01    100
Freq: 15T, dtype: int64

In [3]: rdtools.energy_from_power(power)
   ...: 
C:\Users\KANDERSO\Software\Anaconda3\envs\fleets-dev\lib\site-packages\numpy\lib\nanfunctions.py:1112: RuntimeWarning: Mean of empty slice
  return np.nanmean(a, axis, out=out, keepdims=keepdims)
Traceback (most recent call last):

  File "<ipython-input-3-06a550e25220>", line 1, in <module>
    rdtools.energy_from_power(power)

  File "C:\Users\KANDERSO\Software\Anaconda3\envs\fleets-dev\lib\site-packages\rdtools\normalization.py", line 495, in energy_from_power
    target_frequency = str(int(median_step_ns)) + 'N'

ValueError: cannot convert float NaN to integer
```

**Describe the solution you'd like**
Have `normalization.energy_from_power` use `power.index.freq` if it exists, otherwise fall back to the current approach.

**Describe alternatives you've considered**
None

**Additional context**
I stumbled on this in an edge case when the new AvailabilityAnalysis class found outages that only last one timestamp: https://github.com/NREL/rdtools/blob/development/rdtools/availability.py#L492

**Comment by mdeceglie** [comment]
I think this should be a pretty straightforward change. The median time step is also used for the `max_time_delta`, when `max_time_delta` is not supplied.

**Comment by mdeceglie** [comment]
We will also have to figure out this error for the single-value case:
```python
In [1]: import rdtools
   ...: import pandas as pd
   ...: 
   ...: power = pd.Series([100], index=pd.date_range('2019-01-01', freq='15T', periods=1))
   ...: 
   ...: rdtools.energy_from_power(power, target_frequency='15T', max_timedelta=pd.to_timedelta('15 Minutes'))
   ...: 
/Users/mdecegli/opt/anaconda3/envs/rdtools_dev_mode/lib/python3.7/site-packages/numpy/lib/nanfunctions.py:1111: RuntimeWarning: Mean of empty slice
  return np.nanmean(a, axis, out=out, keepdims=keepdims)
---------------------------------------------------------------------------
UnboundLocalError                         Traceback (most recent call last)
<ipython-input-1-ad829e3c5d9d> in <module>()
      4 power = pd.Series([100], index=pd.date_range('2019-01-01', freq='15T', periods=1))
      5 
----> 6 rdtools.energy_from_power(power, target_frequency='15T', max_timedelta=pd.to_timedelta('15 Minutes'))

/Users/mdecegli/Documents/GitHub/rdtools/rdtools/normalization.py in energy_from_power(power, target_frequency, max_timedelta)
    538     # Set the frequency if we can
    539     try:
--> 540         energy.index.freq = pd.infer_freq(energy.index)
    541     except ValueError:
    542         pass

UnboundLocalError: local variable 'energy' referenced before assignment
```

**Comment by mdeceglie** [comment]
Closed with #214
